### DISCUSSION TITLE
Typescript vs. javascript

### DISCUSSION THREAD

**Discussion by kcramakrishna** [Digital-Assistant/Digital_Assistant_Client#292] [Ideas]
While we move to node, does it make sense to move to typescript ? It may make our code more robust due to the obvious advantages of typescript.
Or how about react + Typescript: https://www.sitepoint.com/react-with-typescript-best-practices/
Apparently React libraries have Typescript support: https://blog.bitsrc.io/5-strong-reasons-to-use-typescript-with-react-bc987da5d907

**Comment by yureshwar** [comment]
How to create a react app using typescript: 

https://rossbulat.medium.com/how-to-use-typescript-with-react-and-redux-a118b1e02b76

**Comment by kcramakrishna** [comment]
1. https://reactjs.org/docs/static-type-checking.html
a. The above link also has support for Kotlin.
b. https://github.com/piotrwitek/react-redux-typescript-guide#react--redux-in-typescript---complete-guide 
c. https://react-typescript-cheatsheet.netlify.app/
d. https://gist.github.com/dilantha111/5ac9db4117e12771a239774ce1944851#file-type-script-support-for-common-libraries-csv

2. https://www.typescriptlang.org/docs/handbook/react.html

Looks like typescript + react is pretty non-controversial. We can use Typescript in react without any fear of incompatibilities. It boils down to preference and convenience.

3. We can use typescript with Reactnative too: https://reactnative.dev/docs/typescript

**Comment by yureshwar** [comment]
Followed this article to create the chrome extension project

https://medium.com/litslink/how-to-create-google-chrome-extension-using-react-js-5c9e343323ff

-- Need to fine tune the application just created a bare bone application and added the eslint configuration from the below 

https://andrebnassis.medium.com/setting-eslint-on-a-react-typescript-project-2021-1190a43ffba

**Comment by yureshwar** [comment]
Our application requires less DOM manipulations.
1. React provides virtual DOM in which the DOM mutations are less
![image](https://user-images.githubusercontent.com/516516/183877036-8ce395d1-46fb-4dff-97fa-b4bb8cd6b5ae.png)

source: https://nordicapis.com/react-vs-angular-which-one-you-should-opt-for/#:~:text=While%20Angular%20is%20responsible%20for,development%20faster%20and%20more%20convenient.

2. React provides more performance over the angular look at the perfomance index in the following link
    https://aglowiditsolutions.com/blog/react-vs-angular/
